src/gentropy/datasource/ensembl/vep_parser.py

```diff
@@ -0,0 +1,769 @@
+
+        Returns:
+            Column: List of dbXrefs for VCV ClinVar identifiers.
+
+        Examples:
+            >>> data = [('VCV2323,RCV3423', 'rs1', 's1',),(None, 'rs1', 's1',),]
+            >>> (
+            ...    spark.createDataFrame(data, ['id', 'rsid', 's'])
+            ...    .groupBy('s')
+            ...    .agg(f.collect_list(f.struct(f.struct(f.split(f.col('id'),',').alias('ClinVar')).alias('var_synonyms'),f.col('rsid').alias('id'),),).alias('cv'),).select(VariantEffectPredictorParser._extract_clinvar_xrefs(f.col('cv')).alias('dbXrefs'))
+            ...    .show(truncate=False)
+            ... )
+            +--------------------+
+            |dbXrefs             |
+            +--------------------+
+            |[{VCV2323, clinVar}]|
+            +--------------------+
+            <BLANKLINE>
+        """
+        variants_w_clinvar_ref = f.filter(
+            colocated_variants,
+            lambda variant: variant.getItem("var_synonyms")
+            .getItem("ClinVar")
+            .isNotNull(),
+        )
+
+        clin_var_ids = f.transform(
+            variants_w_clinvar_ref,
+            lambda var: f.filter(
+                var.getItem("var_synonyms").getItem("ClinVar"),
+                lambda x: x.startswith("VCV"),
+            ),
+        )
+
+        return VariantEffectPredictorParser._generate_dbxrefs(
+            f.flatten(clin_var_ids), "clinVar"
+        )
+
+    @staticmethod
+    def _get_most_severe_transcript(
+        transcript_column_name: str, field_name: str
+    ) -> Column:
+        """Return transcript with the most highest in silico predicted score.
+
+        This method assumes the higher the score, the more severe the consequence of the variant is.
+        Hence, by selecting the transcript with the highest score, we are selecting the most severe consequence.
+
+        Args:
+            transcript_column_name (str): Name of the column containing the list of transcripts.
+            field_name (str): Name of the field containing the severity score.
+
+        Returns:
+            Column: Most severe transcript.
+
+        Examples:
+            >>> data = [("v1", 0.2, 'transcript1'),("v1", None, 'transcript2'),("v1", 0.6, 'transcript3'),("v1", 0.4, 'transcript4'),]
+            >>> (
+            ...    spark.createDataFrame(data, ['v', 'score', 'transcriptId'])
+            ...    .groupBy('v')
+            ...    .agg(f.collect_list(f.struct(f.col('score'),f.col('transcriptId'))).alias('transcripts'))
+            ...    .select(VariantEffectPredictorParser._get_most_severe_transcript('transcripts', 'score').alias('most_severe_transcript'))
+            ...    .show(truncate=False)
+            ... )
+            +----------------------+
+            |most_severe_transcript|
+            +----------------------+
+            |{0.6, transcript3}    |
+            +----------------------+
+            <BLANKLINE>
+        """
+        assert isinstance(
```

Comment:
It might not be necessary, but I've been making that mistake when I'm passing a column instead of a column name. I don't really like the `order_array_of_structs_by_field` function, because it take a name of a column, and returns a column. It keeps confusing me.